ledger-core/virtual/integration/deactivate_test.go

```diff
@@ -401,3 +398,108 @@ func TestVirtual_CallMethod_On_DeactivatedDirtyState(t *testing.T) {
 
 	mc.Finish()
 }
+
+func TestVirtual_CallDeactivate_Intolerable(t *testing.T) {
+	t.Log("C5469")
+
+	table := []struct {
+		name  string
+		state contract.StateFlag
+	}{
+		{name: "ValidatedState", state: contract.CallValidated},
+		{name: "DirtyState", state: contract.CallDirty},
+	}
+
+	for _, testCase := range table {
+		t.Run(testCase.name, func(t *testing.T) {
+			defer testutils.LeakTester(t)
+
+			mc := minimock.NewController(t)
+
+			server, ctx := utils.NewUninitializedServer(nil, t)
+			defer server.Stop()
+
+			runnerMock := logicless.NewServiceMock(ctx, mc, func(execution execution.Context) string {
+				return execution.Request.CallSiteMethod
+			})
+			server.ReplaceRunner(runnerMock)
+			server.Init(ctx)
+
+			var (
+				class        = gen.UniqueGlobalRef()
+				objectGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+				prevPulse    = server.GetPulse().PulseNumber
+			)
+
+			// Create object
+			{
+				server.IncrementPulse(ctx)
+
+				report := &payload.VStateReport{
+					Status: payload.Ready,
+					Object: objectGlobal,
+					AsOf:   prevPulse,
+					ProvidedContent: &payload.VStateReport_ProvidedContentBody{
+						LatestDirtyState: &payload.ObjectState{
+							Reference: reference.Local{},
+							Class:     class,
+							State:     []byte("initial state"),
+						},
+						LatestValidatedState: &payload.ObjectState{
+							Reference: reference.Local{},
+							Class:     class,
+							State:     []byte("initial state"),
+						},
+					},
+				}
+				wait := server.Journal.WaitStopOf(&handlers.SMVStateReport{}, 1)
+				server.SendPayload(ctx, report)
+				testutils.WaitSignalsTimed(t, 10*time.Second, wait)
+			}
+
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+
+			// Add VCallResult check
+			{
+				typedChecker.VCallResult.Set(func(result *payload.VCallResult) bool {
+					contractErr, sysErr := foundation.UnmarshalMethodResult(result.ReturnArguments)
+					require.NoError(t, sysErr)
+					require.Equal(t, "intolerable call trying to change object state", contractErr.Error())
+
+					return false
+				})
+			}
+
+			// Add executor mock for method `Destroy`
+			{
+				requestResult := requestresult.New([]byte("done"), objectGlobal)
+				requestResult.SetDeactivate(descriptor.NewObject(objectGlobal, server.RandomLocalWithPulse(), class, []byte("initial state"), false))
+				runnerMock.AddExecutionMock("Destroy").AddStart(func(ctx execution.Context) {},
+					&execution.Update{
+						Type:   execution.Done,
+						Result: requestResult,
+					},
```

Comment:
deactivation runner mock must be more complex
```
runnerMock.AddExecutionMock(deactivateMethodName).AddStart(
			nil,
			&execution.Update{
				Type:     execution.OutgoingCall,
				Result:   requestResult,
				Outgoing: execution.NewRPCBuilder(outgoing, objectRef).Deactivate(),
			},
		).AddContinue(
			nil,
			&execution.Update{
				Type:   execution.Done,
				Result: requestresult.New([]byte("finish Deactivate"), objectRef),
			},
		)
```
i think that better to use right mock because it can be used as  example

Reply:
fixed